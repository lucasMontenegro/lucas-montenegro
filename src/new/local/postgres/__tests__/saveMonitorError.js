const mockFn = jest.fn()
jest.mock(`os`, () => ({ EOL: `'os.EOL'` }))
jest.mock(`fs`, () => ({}))
require("fs").appendFile = (...args) => {
  mockFn(`fs.appendFile`, args)
}
jest.mock(`new/local/utils/simpleErrorCallback`, () => ({
  default: `new/local/utils/simpleErrorCallback`,
}))
jest.mock(`new/local/utils/globals`, () => ({
  default: {
    process: { env: { NODE_PATH: `/nodePath` } },
  },
}))
const { default: saveMonitorError } = require("new/local/postgres/saveMonitorError")
describe(`new/local/postgres/saveMonitorError`, () => {
  afterEach(() => mockFn.mockClear())
  test.each([
    [true],
    [false],
  ])(`should save the error message (info.time %j)`, time => {
    const info = { time }
    saveMonitorError(`msg`, info)
    expect({ info, mockFnCalls: mockFn.mock.calls }).toMatchSnapshot()
  })
})