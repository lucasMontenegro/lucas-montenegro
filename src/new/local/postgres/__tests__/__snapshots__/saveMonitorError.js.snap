// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/local/postgres/saveMonitorError should save the error message (info.time false) 1`] = `
Object {
  "info": Object {
    "time": false,
  },
  "mockFnCalls": Array [
    Array [
      "fs.appendFile",
      Array [
        "/pg-errors.log",
        "'os.EOL'msg",
        "new/local/utils/simpleErrorCallback",
      ],
    ],
  ],
}
`;

exports[`new/local/postgres/saveMonitorError should save the error message (info.time true) 1`] = `
Object {
  "info": Object {
    "time": true,
  },
  "mockFnCalls": Array [
    Array [
      "fs.appendFile",
      Array [
        "/pg-errors.log",
        "'os.EOL''os.EOL'msg",
        "new/local/utils/simpleErrorCallback",
      ],
    ],
  ],
}
`;
